src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -447,6 +447,7 @@ private BoundForEachStatement BindForEachPartsWorker(BindingDiagnosticBag diagno
             else
             {
                 ReportDiagnosticsIfObsolete(diagnostics, elementConversion, _syntax.ForEachKeyword, hasBaseReceiver: false);
+                CheckConstraintLanguageVersionAndRuntimeSupportForConversion(_syntax.ForEachKeyword, elementConversion, diagnostics);
```

Comment:
I didn't see tests in `StaticAbstractMembersInInterfacesTests.cs` for this code path. It seems like the test would be a foreach loop which subjects the collection element to an abstract static conversion, and either the language version or runtime doesn't support the usage of the conversion.

Reply:
> I didn't see tests in StaticAbstractMembersInInterfacesTests.cs for this code path. It seems like the test would be a foreach loop which subjects the collection element to an abstract static conversion, and either the language version or runtime doesn't support the usage of the conversion.

Testing this is tracked by [https://github.com/dotnet/roslyn/issues/53870](https://github.com/dotnet/roslyn/issues/53870). Ther are a lot of similar scenarios to cover, I am not planning to do this in this PR, but I am pretty comfortable to add obvious checks that are working for other tested scenarios.